sim_pipeline/gg_lens_pop.py

```diff
@@ -72,13 +74,18 @@ def draw_population(self):
         # Estimate the number of lensing systems
         num_lenses = self._lens_galaxies.deflector_number()
         num_sources = self._source_galaxies.galaxies_number()
-        num_lens_systems = int(num_lenses * num_sources)
-        print(num_lens_systems)
-        for _ in range(100):
-            source = self._source_galaxies.draw_galaxy()
+        num_sources_tested_mean = (testarea * self.f_sky * num_sources)/12960000
+        print(num_sources_tested_mean)
+        print(np.int(num_lenses*num_sources_tested_mean))
+
+# Draw a population of galaxy-galaxy lenses within the area.
+        for _ in range(num_lenses):
             lens = self._lens_galaxies.draw_deflector()
-            gg_lens = GGLens(deflector_dict=lens, source_dict=source, cosmo=self.cosmo)
-            # Check the validity of the lens system
-            if gg_lens.validity_test():
-                gg_lens_population.append(gg_lens)
+            for _ in range(max(1,int(np.rint(np.random.normal
```

Comment:
also here, it is unclear what is the range. Perhaps have a separate function/line of code defining the length of this for-loop and comment on how you calculate it